packages/editor/src/extensions/code-block/highlighter.ts

```diff
@@ -121,10 +121,17 @@ export function HighlighterPlugin({
         },
         async update(view) {
           const pluginState = HIGHLIGHTER_PLUGIN_KEY.getState(view.state);
+          const pluginLanguages = pluginState?.languages;
+          const registeredLanguage = refractor.listLanguages();
           if (!pluginState) return;
 
           const changedBlocks: Set<string> = new Set();
           for (const blockKey in pluginState.languages) {
+            if (pluginLanguages) {
+              if (!registeredLanguage.includes(pluginLanguages[blockKey])) {
+                HIGHLIGHTED_BLOCKS.delete(blockKey);
+              }
+            }
```

Comment:
Simplify this and merge it with the if statement below. You might need to move the `if..statement` a few downs lower.